inc/classes/Db.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/*  Poweradmin, a friendly web-based admin tool for PowerDNS.
+ *  See <http://www.poweradmin.org> for more details.
+ *
+ *  Copyright 2007-2009  Rejo Zenger <rejo@zenger.nl>
+ *  Copyright 2010-2016  Poweradmin Development Team
+ *      <http://www.poweradmin.org/credits.html>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Poweradmin;
+
+use Doctrine\DBAL\DriverManager;
+
+/**
+ * Db class
+ *
+ * @package     Poweradmin
+ * @copyright   2007-2010 Rejo Zenger <rejo@zenger.nl>
+ * @copyright   2010-2016 Poweradmin Development Team
+ * @license     http://opensource.org/licenses/GPL-3.0 GPL
+ */
+class Db
+{
+    /**
+     * Internal storage of database-connections
+     * @var array
+     */
+    protected static $connections = array();
+
+
+    /**
+     * Creates database-connection using Doctrine (DBAL)
+     *
+     * @param array $parameters
+     * @param string $name
+     * @return bool
+     */
+    public static function createConnection($parameters, $name = 'default')
+    {
+        // Check for driver in parameters
+        if (!is_array($parameters) || !array_key_exists('driver', $parameters)) {
```

Comment:
Implemented method, in this case it will not replace form validation but its another security-step.